builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -61,9 +91,14 @@ func resourceAwsS3BucketObjectPut(d *schema.ResourceData, meta interface{}) erro
 
 	resp, err := s3conn.PutObject(
 		&s3.PutObjectInput{
-			Bucket: aws.String(bucket),
-			Key:    aws.String(key),
-			Body:   file,
+			Bucket:             aws.String(bucket),
+			Key:                aws.String(key),
+			Body:               file,
+			CacheControl:       aws.String(cacheControl),
+			ContentDisposition: aws.String(contentDisposition),
+			ContentEncoding:    aws.String(encoding),
+			ContentLanguage:    aws.String(contentLanguage),
+			ContentType:        aws.String(contentType),
```

Comment:
Because these fields are optional, `d.Get("content_encoding").(string)` will return an empty string `""` if it's not included in the configuration file. We'll then pass that on in the `PutObjectIntput` struct, so `CacheControl` will equal `""`. In most of the AWS API calls, this is not correct, and can introduce unexpected or even undefined behavior. 

We should check if the values are `""` and conditionally add them to the struct